application/routers/entity.py

```diff
@@ -276,6 +285,12 @@ def search_entities(
         if params.get("field") is not None:
             include = set([to_snake(field) for field in params.get("field")])
             entities = _get_entity_json(data["entities"], include=include)
+        elif exclude_field:
```

Comment:
You won't need to convert the object back to a data model if you remove the code from get_entity_search as that's what's causing it not to be an entity model.